app_test.go

```diff
@@ -178,6 +179,61 @@ func Test_App_Errors(t *testing.T) {
 	}
 }
 
+type customConstraint struct{}
+
+func (*customConstraint) Name() string {
+	return "test"
+}
+
+func (*customConstraint) Execute(param string, args ...string) bool {
+	if param == "test" && len(args) == 1 && args[0] == "test" {
+		return true
+	}
+
+	if len(args) == 0 && param == "c" {
+		return true
+	}
+
+	return false
+}
+
+func Test_App_CustomConstraint(t *testing.T) {
```

Comment:
please add a benchmark for a custom contraints which is simple so that we can test the runtime performance
and later get to know if it gets worse or if we could actively optimize it

Reply:
> please add a benchmark for a custom contraints which is simple so that we can test the runtime performance and later get to know if it gets worse or if we could actively optimize it

Do you mean adding a benchmark using RoutePatternMatch?

Reply:
@efectn yes so that we can test the path with the custom constraint

Reply:
> @efectn yes so that we can test the path with the custom constraint

So we need to add 3rd parameter to the function?

Reply:
And either config or constraints parameter should be variadic

Reply:
all good, we can add it later if it makes sense, the custom constraints are only used in edge cases and should have almost the same performance as the other process where we already have a benchmark